ftw/keywordwidget/widget.py

```diff
@@ -313,7 +313,7 @@ def addItem(idx, term, prefix=''):
                  'selected': selected})
 
         if self.async:
-            terms = [self.terms.getTermByToken(v) for v in self.value]
+            terms = [self.terms.getTermByToken(v) for v in self.value if v]
```

Comment:
What exactly is in the `self.value` list, are those tokens of terms? Or is it the raw submitted value?
Does this change prevent us from using a falsy term-value (`''`, `None`, `False`, `0`)?
If so, would it help to test for an empty string explicitly here?

Reply:
This are the tokens from the terms selected in the widget.

Yes. The `getTermByToken` method have to raise a lookup-error if it can't find a term by definition. In some cases, we get this list with an empty string `['']` and it will fail.

I think, we never want to get a term through a falsy value because it will always raise an error.

@jone 👍 thats true, I could make a more explicitly test here.

Reply:
I think that's ok here: the term-token is not the term-value. We need to be able to build vocabularies with falsy values (e.g. "Nothing selected"), but I think we can demand that all tokens are truthy. 👍